lumeer-storage/lumeer-storage-api/src/main/java/io/lumeer/storage/api/dao/UserDao.java

```diff
@@ -18,28 +18,23 @@
  */
 package io.lumeer.storage.api.dao;
 
-import io.lumeer.api.model.Organization;
 import io.lumeer.api.model.User;
 
 import java.util.List;
 import java.util.Optional;
 
 public interface UserDao {
 
-   void createUsersRepository(Organization organization);
+   User createUser(String organizationId, User user);
 
-   void deleteUsersRepository(Organization organization);
+   User updateUser(String organizationId, String userId, User user);
 
-   void setOrganization(Organization organization);
+   void deleteUser(String organizationId, String userId);
 
-   User createUser(User user);
+   void deleteGroupFromUsers(String organizationId, String group);
 
-   void updateUser(String id, User user);
+   Optional<User> getUserByEmail(String organizationId, String email);
```

Comment:
We don't really need to know `organizationId` in order to get a user by email, do we?

Reply:
We need to know `organizationId` to obtain correct groups for user.

Reply:
Oh, I see. I'm just wondering if we cannot do this kind of logic in the facade. It's what facades are there for. I know we would read more data from the database but how many organizations will users be in? One, two?

I think the same principle is used with permissions which are filtered on the facade layer.

The other think is that we cache the user so what happens if he switch the organization within the same session? I'm afraid the necessary data will not be in the cache since they were filtered by DAO layer.